projects/plugins/jetpack/extensions/blocks/subscriptions/view.js

```diff
@@ -36,6 +36,41 @@ function show_iframe( data ) {
 	return showModal( url );
 }
 
+function show_iframe_get_current_user_email( blog, emailInput ) {
+	const existingIframe = document.getElementById( 'jp-subscribe-current-user-email-iframe' );
+	if ( existingIframe ) {
+		return;
+	}
+	const params = new URLSearchParams( {
+		blog,
+		plan: 'newsletter',
+		source: 'jetpack_subscribe_current_user_email',
+		jwt_token: getTokenFromCookie(),
+	} );
+	const url = 'https://subscribe.wordpress.com/memberships/?' + params.toString();
+
+	const iframe = document.createElement( 'iframe' );
+	iframe.setAttribute( 'id', 'jp-subscribe-current-user-email-iframe' );
+	iframe.style.display = 'none';
+	iframe.src = url;
+
+	document.body.appendChild( iframe );
+
+	const handleIframeMessage = message => {
+		if (
+			message.origin === 'https://subscribe.wordpress.com' &&
+			message.data.action === 'current_user_email' &&
+			message.data.email
```

Comment:
data.action and data.email might not exist and cause an error. Maybe we should use optional chaining instead?

Reply:
https://github.com/Automattic/jetpack/pull/39633/commits/270f8b982c8424978a3b6106bb90bdd41db09cf5